Code/Engine/Foundation/Memory/Implementation/MemoryUtils_inl.h

```diff
@@ -590,6 +596,42 @@ inline void ezMemoryUtils::Prepend(T* pDestination, T&& source, size_t uiCount,
   }
 }
 
+template <typename T>
+EZ_ALWAYS_INLINE void ezMemoryUtils::Prepend(T* pDestination, const T* source, size_t uiSourceCount, size_t uiCount, ezTypeIsPod)
+{
+  memmove(pDestination + uiSourceCount, pDestination, uiCount * sizeof(T));
+  CopyConstructArray(pDestination, source, uiSourceCount, ezTypeIsPod());
+}
+
+template <typename T>
+EZ_ALWAYS_INLINE void ezMemoryUtils::Prepend(T* pDestination, const T* source, size_t uiSourceCount, size_t uiCount, ezTypeIsMemRelocatable)
+{
+  memmove(pDestination + 1, pDestination, uiCount * sizeof(T));
+  CopyConstructArray(pDestination, source, uiSourceCount, ezTypeIsClass());
+}
+
+template <typename T>
+inline void ezMemoryUtils::Prepend(T* pDestination, const T* source, size_t uiSourceCount, size_t uiCount, ezTypeIsClass)
+{
+  EZ_CHECK_CLASS(T);
+
+  if (uiCount > 0)
+  {
+    MoveConstruct(pDestination + uiCount, std::move(pDestination[uiCount - uiSourceCount]), uiCount);
+
+    for (size_t i = uiCount - 1; i-- > 0;)
+    {
+      pDestination[i + 1] = std::move(pDestination[i]);
```

Comment:
this shifts each element only by one(from i to i+1), but it should shift it by uiSourceCount

Reply:
Fixed